src/main/java/loqor/ait/core/lock/LockedDimension.java

```diff
@@ -0,0 +1,72 @@
+package loqor.ait.core.lock;
+
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.JsonOps;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+
+import net.minecraft.item.ItemStack;
+import net.minecraft.registry.RegistryKey;
+import net.minecraft.registry.RegistryKeys;
+import net.minecraft.text.Text;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.dynamic.Codecs;
+
+import loqor.ait.AITMod;
+import loqor.ait.api.Nameable;
+import loqor.ait.core.util.WorldUtil;
+import loqor.ait.data.Loyalty;
+import loqor.ait.registry.unlockable.Unlockable;
+
+public record LockedDimension(Identifier dimension, ItemStack stack) implements Unlockable, Nameable {
+    public static final Codec<LockedDimension> CODEC = Codecs.exceptionCatching(RecordCodecBuilder.create(instance -> instance.group(
+            Identifier.CODEC.fieldOf("dimension").forGetter(LockedDimension::dimension),
+            ItemStack.CODEC.fieldOf("stack").forGetter(LockedDimension::stack))
+            .apply(instance, LockedDimension::new)));
+
+    @Override
+    public Identifier id() {
+        return this.dimension();
+    }
+
+    public static LockedDimension fromInputStream(InputStream stream) {
+        return fromJson(JsonParser.parseReader(new InputStreamReader(stream)).getAsJsonObject());
+    }
+
+    public static LockedDimension fromJson(JsonObject json) {
+        AtomicReference<LockedDimension> created = new AtomicReference<>();
+
+        CODEC.decode(JsonOps.INSTANCE, json).get().ifLeft(planet -> created.set(planet.getFirst())).ifRight(err -> {
+            created.set(null);
+            AITMod.LOGGER.error("Error decoding datapack locked dim: {}", err);
+        });
+
+        return created.get();
+    }
+
+    @Override
+    public UnlockType unlockType() {
+        return UnlockType.DIMENSION;
+    }
+
+    @Override
+    public Optional<Loyalty> requirement() {
+        return Optional.of(Loyalty.fromLevel(100));
```

Comment:
So wait are they supposed to all unlock on loyalty 100 or whar

Reply:
nope and they don't because i didn't add it to the loyalty check
i just put that because it needed a loyalty value

Reply:
Thats confusing
Either stop using the unlockable interface here
OR
Rewrite unlockables to have a check method with a generic instead of the requirement

Reply:
those both sound hard